diff --git a/config.def.h b/config.def.h
index 1edb647..6edb657 100644
--- a/config.def.h
+++ b/config.def.h
@@ -1,6 +1,7 @@
 /* See LICENSE file for copyright and license details. */
 /* Default settings; can be overriden by command line. */
 
+static int exact = 0;						/* -e  option; if 1, dmenu only matches exactly */
 static int topbar = 1;                      /* -b  option; if 0, dmenu appears at bottom     */
 /* -fn option overrides fonts[0]; default X11 font or font set */
 static const char *fonts[] = {
diff --git a/dmenu.c b/dmenu.c
index 818313a..ec1bd2c 100644
--- a/dmenu.c
+++ b/dmenu.c
@@ -258,7 +258,7 @@ match(void)
 			appenditem(item, &matches, &matchend);
 		else if (!fstrncmp(tokv[0], item->text, len))
 			appenditem(item, &lprefix, &prefixend);
-		else
+		else if (!exact)
 			appenditem(item, &lsubstr, &substrend);
 	}
 	if (lprefix) {
@@ -728,6 +728,8 @@ main(int argc, char *argv[])
 			exit(0);
 		} else if (!strcmp(argv[i], "-b")) /* appears at the bottom of the screen */
 			topbar = 0;
+		else if (!strcmp(argv[i], "-e"))   /* only exact matching */
+			exact = 1;
 		else if (!strcmp(argv[i], "-f"))   /* grabs keyboard before reading stdin */
 			fast = 1;
 		else if (!strcmp(argv[i], "-i")) { /* case-insensitive item matching */
